src/main/java/client/http5/OpenSearchRestClientImpl.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package client.http5;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsRequest;
+import org.opensearch.action.admin.indices.settings.get.GetSettingsRequest;
+import org.opensearch.action.admin.indices.settings.get.GetSettingsResponse;
+import org.opensearch.action.search.*;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.indices.CreateIndexRequest;
+import org.opensearch.client.indices.GetIndexRequest;
+import org.opensearch.client.indices.GetIndexResponse;
+import org.opensearch.client.indices.GetMappingsRequest;
+import org.opensearch.client.indices.GetMappingsResponse;
+import org.opensearch.cluster.metadata.AliasMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.search.builder.PointInTimeBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.search.sort.SortOrder;
+import org.opensearch.sql.opensearch.client.OpenSearchClient;
+import org.opensearch.sql.opensearch.mapping.IndexMapping;
+import org.opensearch.sql.opensearch.request.OpenSearchQueryRequest;
+import org.opensearch.sql.opensearch.request.OpenSearchRequest;
+import org.opensearch.sql.opensearch.request.OpenSearchScrollRequest;
+import org.opensearch.sql.opensearch.response.OpenSearchResponse;
+import org.opensearch.transport.client.node.NodeClient;
+
+/**
+ * OpenSearch REST client to support standalone mode that runs entire engine from remote.
+ *
+ * <p>TODO: Support for authN and authZ with AWS Sigv4 or security plugin.
+ */
+public class OpenSearchRestClientImpl implements OpenSearchClient {
+
+  /** OpenSearch high level REST client. */
+  private final RestHighLevelClient client;
+
+  public OpenSearchRestClientImpl(RestHighLevelClient client) {
+    this.client = client;
+  }
+
+  @Override
+  public boolean exists(String indexName) {
+    System.out.println("OpenSearchRestClientImpl.exists()");
```

Comment:
Any logger available? Usually we avoid this in Java project. Could you check if any sysout or main method can be cleaned up?